app/src/main/java/ch/epfl/sweng/studyup/AuthenticationActivity.java

```diff
@@ -0,0 +1,107 @@
+package ch.epfl.sweng.studyup;
+
+import android.net.Uri;
+import android.os.StrictMode;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Scanner;
+import org.json.*;
+
+public class AuthenticationActivity extends AppCompatActivity {
+
+    private void reportAuthError(String error) {
+
+        Log.e(getString(R.string.auth_error), error);
+        Toast.makeText(AuthenticationActivity.this,
+                "Unable to authenticate.",
+                Toast.LENGTH_SHORT).show();
+    }
+
+    private String getTokenFromResponse(String response) throws JSONException {
+        JSONObject tokenResponseJSON = new JSONObject(response);
+
+        if (tokenResponseJSON.has("error")) {
+            String error = tokenResponseJSON.getString("error");
+            reportAuthError(error);
+            return null;
+        }
+        else {
+            String token = tokenResponseJSON.getString("access_token");
+            return token;
+        }
+    }
+
+    private String getToken(String code) throws IOException, JSONException {
+
+        String tokenURL = "https://studyup-authenticate.herokuapp.com/getToken" + "?code=" + code;
+        HttpURLConnection connection = (HttpURLConnection) new URL(tokenURL).openConnection();
+        InputStream stream = connection.getInputStream();
+        String tokenResponse = new Scanner(stream).useDelimiter("\\A").next();
+
+        return getTokenFromResponse(tokenResponse);
+    }
+
+    private void displayProfileDataFromResponse(String response) throws JSONException {
+
+        JSONObject profileResponseJSON = new JSONObject(response);
+        if (profileResponseJSON.has("error")) {
```

Comment:
Isn't it duplicated code compared to just what is before? Helper method checkJSONError?